buffer/src/main/java/io/netty/buffer/PooledByteBuf.java

```diff
@@ -98,6 +98,11 @@ public final ByteBuf capacity(int newCapacity) {
             return this;
         }
         checkNewCapacity(newCapacity);
+        if (isDirect()) {
+            chunk.arena.parent.incrementUsedDirectBytes(newCapacity - length);
+        } else {
+            chunk.arena.parent.incrementUsedHeapBytes(newCapacity - length);
+        }
```

Comment:
This logic should move into `PoolArena.reallocate`.
At this location, if the new capacity is less than the current capacity, we would incorrectly count the used memory down even though the memory wasn't actually returned to the pool.